### PR TITLE
avm2: Added all missing `[API()]` to classes from the list

### PR DESCRIPTION
**Pull request by MartySVK**
Dinnerbone's list - https://gist.github.com/Dinnerbone/0cfad60f4587292005d41e566368c709

Plus some others I discovered randomly.

Hope I didn't make a mistake anywhere.

### CODE REVIEW

File: core/src/avm2/globals/flash/display3D/Context3DBufferUsage.as [hunk 1]
```diff
@@ -6,6 +6,7 @@
 package flash.display3D
 {
 
+    [API("692")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/display3D/Context3DBufferUsage.as:9]: I think this one should be `694`, for Flash Player 12: https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3DBufferUsage.html

File: core/src/avm2/globals/flash/system/WorkerDomain.as [hunk 2]
```diff
@@ -1,4 +1,5 @@
 package flash.system {
+    [API("680")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/system/WorkerDomain.as:2]: I think this should be `682` for Flash Player 11.4: https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/WorkerDomain.html

File: core/src/avm2/globals/flash/events/GameInputEvent.as [hunk 3]
```diff
@@ -1,4 +1,5 @@
 package flash.events {
+    [API("688")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/events/GameInputEvent.as:2]: I think this is AIR-only (well, according to the docs), so should be `689`?
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/events/GameInputEvent.html

But I feel like we've been here before, and this is one of the many cases where the docs just lie.

File: core/src/avm2/globals/flash/events/ThrottleType.as [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package flash.events
 {
 
+    [API("676")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/events/ThrottleType.as:4]: I think this should be `674` for Flash Player 11:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/events/ThrottleType.html

File: core/src/avm2/globals/flash/events/ThrottleEvent.as [hunk 5]
```diff
@@ -5,6 +5,7 @@
 package flash.events
 {
     
+    [API("676")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/events/ThrottleEvent.as:8]: I think this should be `674` for Flash Player 11:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/events/ThrottleEvent.html

File: core/src/avm2/globals/flash/profiler/Telemetry.as [hunk 6]
```diff
@@ -1,4 +1,5 @@
 package flash.profiler {
+    [API("678")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/profiler/Telemetry.as:2]: I think this should be `682`, for Flash Player 11.4:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/profiler/Telemetry.html

File: core/src/avm2/globals/flash/media/StageVideoAvailabilityReason.as [hunk 7]
```diff
@@ -6,6 +6,7 @@
 package flash.media
 {
 
+    [API("688")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/media/StageVideoAvailabilityReason.as:9]: I think this should be `670` for Flash Player 10.2:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/StageVideoAvailabilityReason.html

File: core/src/avm2/globals/flash/media/AudioOutputChangeReason.as [hunk 8]
```diff
@@ -6,6 +6,7 @@
 package flash.media
 {
 
+    // TODO: [API("724")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/media/AudioOutputChangeReason.as:9]: I think mentioning Flash Player 27, and that `724` is just a guess/extrapolation, would be better.

File: core/src/avm2/globals/flash/events/AudioOutputChangeEvent.as [hunk 9]
```diff
@@ -4,6 +4,7 @@
 // It won't be regenerated in the future, so feel free to edit and/or fix
 package flash.events
 {
+    // TODO: [API("724")]
```
Comment (torokati44) [inline core/src/avm2/globals/flash/events/AudioOutputChangeEvent.as:7]: I think mentioning Flash Player 27, and that `724` is just a guess/extrapolation, would be better.

File: core/src/avm2/globals/flash/display3D/Context3DWrapMode.as [hunk 10]
```diff
@@ -6,18 +6,21 @@
...
         // Clamp in U axis but Repeat in V axis.
+        [API("696")]
         public static const CLAMP_U_REPEAT_V:String = "clamp_u_repeat_v";
 
         // Repeat (tile) texture coordinates outside the 0..1 range.
         public static const REPEAT:String = "repeat";
 
         // Repeat in U axis but Clamp in V axis.
+        [API("696")]
         public static const REPEAT_U_CLAMP_V:String = "repeat_u_clamp_v";
```
Comment (torokati44) [inline core/src/avm2/globals/flash/display3D/Context3DWrapMode.as:24]: The class-level annotation looks right, but the constants don't seem to have their Flash Player versions mentioned in the docs:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3DWrapMode.html

File: core/src/avm2/globals/flash/display3D/Context3DProfile.as [hunk 11]
```diff
@@ -15,15 +16,19 @@ package flash.display3D
...
+        [API("698")]
         public static const STANDARD:String = "standard";
 
         // Use an standard profile to target GPUs which support AGAL2 and float textures.
+        [API("702")]
         public static const STANDARD_CONSTRAINED:String = "standardConstrained";
 
         // Use standard extended profile to target GPUs which support AGAL3 and instanced drawing feature.
+        [API("704")]
         public static const STANDARD_EXTENDED:String = "standardExtended";
```
Comment (torokati44) [inline core/src/avm2/globals/flash/display3D/Context3DProfile.as:32]: I think this should be `706` for Flash Player 18:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3DProfile.html#STANDARD_EXTENDED

**Comment by evilpie** [comment]
So I think these are cases where what is defined in the playerglobal doesn't match what the documentation says, right @torokati44 ? We should just add a comment to those and merge this.

**Comment by torokati44** [comment]
Added these as comments, the rest of my notes can be dismissed:

image

**Review by torokati44 — APPROVED** [review]
Comparing to `playerglobal.swc`, no mistakes were spotted. Thank you, @MartySVK, and sorry again for the noise!

**Comment by torokati44** [comment]
I'd be curious what your difference-detector script says with this in, @Dinnerbone.

**Review by Lord-McSweeney — APPROVED** [review]

**Comment by MartySVK** [comment]
> Comparing to `playerglobal.swc`, no mistakes were spotted. Thank you, @MartySVK, and sorry again for the noise!

no problem and thanks for the review

**Pull request merged**